lib/chai/core/assertions.js

```diff
@@ -1532,6 +1532,56 @@ module.exports = function (chai, _) {
   });
 
   /**
+   * ### .oneOf(list)
+   *
+   * Assert that a value appears somewhere in the top level of array `list`.
+   *
+   *     expect('a').to.be.oneOf(['a', 'b', 'c']);
+   *     expect(9).to.not.be.oneOf(['z']);
+   *     expect([3]).to.not.be.oneOf([1, 2, [3]]);
+   *
+   *     var three = [3];
+   *     // for object-types, contents are not compared
+   *     expect(three).to.not.be.oneOf([1, 2, [3]]);
+   *     // comparing references works
+   *     expect(three).to.be.oneOf([1, 2, three]);
+   *
+   * @name oneOf
+   * @param {Array<*>} list
+   * @param {String} message _optional_
+   * @api public
+   */
+
+  function oneOf (list, msg) {
+    if (msg) flag(this, 'message', msg);
+    var expected = flag(this, 'object');
+    new Assertion(list).to.be.an('array');
+    var assertion = function () {
+        var matched = false;
+        // without a `deep` flag set, only examine top-level members. Do not recurse.
+        list.forEach(function (topLevelMember) {
+            // must be an exact reference to match. There are no comparisons for content!
+            if (topLevelMember === expected) {
+                matched = true;
+                return false;
+            }
+        });
+        return matched;
+    };
```

Comment:
We didn't have this function in the previous commits did we? I'm not sure it is needed. `indexOf` should work just as well here, no?

(from http://www.ecma-international.org/ecma-262/6.0/index.html#sec-array.prototype.indexof)
> indexOf compares searchElement to the elements of the array, in ascending order, using the Strict Equality Comparison algorithm (7.2.13), and if found at one or more indices, returns the smallest such index; otherwise, −1 is returned.

I think having `list.indexOf(expected) !== -1` would work just as well. What do you think?